packages/blocks/src/page-block/default/utils.ts

```diff
@@ -0,0 +1,70 @@
+import type { BaseBlockModel } from '@blocksuite/store';
+import { assertExists, getBlockById } from '../../__internal__/utils';
+
+export const pick = (blocks: BaseBlockModel[], x: number, y: number) => {
+  let result = null;
+  for (let index = 0; index <= blocks.length - 1; index++) {
+    if (blocks[index].flavour === 'affine:embed') {
+      const hoverDom = getBlockById(blocks[index].id);
+      const hoverImage = hoverDom?.querySelector('img');
+      const imageRect = hoverImage?.getBoundingClientRect();
+      assertExists(imageRect);
+      if (isPointIn(imageRect, x, y)) {
+        result = {
+          position: {
+            x: imageRect.right + 10,
+            y: imageRect.top,
+          },
+          model: blocks[index],
+        };
+      }
+    }
+  }
+  return result;
+};
+
+function isPointIn(block: DOMRect, x: number, y: number): boolean {
+  if (
+    x < block.left ||
+    x > block.left + block.width + 50 ||
+    y < block.top ||
+    y > block.top + block.height
+  ) {
+    return false;
+  }
+  return true;
+}
+
+export function downloadImage(url: string | undefined) {
```

Comment:
the `undefined` should not exist in the type of input, because it always fails if the input is undefined.

So the `assertExists` can be safely removed.